### PR TITLE
R2RDump - Ignore sensitive properties to pass tests

### PR DESCRIPTION
**Pull request by acmyu**
Fix for https://github.com/dotnet/coreclr/issues/19011

### CODE REVIEW

**Comment by acmyu** [comment]
@dotnet-bot test Ubuntu x64 Checked CoreFX Tests please

**Comment by acmyu** [comment]
@dotnet-bot test OSX10.12 x64 Checked CoreFX Tests please

**Comment by acmyu** [comment]
@dotnet-bot test x64_checked_windows_nt_tailcallstress please

**Comment by acmyu** [comment]
@dotnet-bot test OSX10.12 x64 Checked CoreFX Tests please
@dotnet-bot test Windows_NT x86 full_opt ryujit CoreCLR Perf Tests Correctness please

**Comment by acmyu** [comment]
@dotnet-bot test OSX10.12 x64 Checked CoreFX Tests please

File: src/tools/r2rdump/R2RSection.cs [hunk 1]
```diff
@@ -42,8 +42,13 @@ public enum SectionType
         /// </summary>
         public int Size { get; set; }
 
-        public R2RSection(SectionType type, int rva, int size)
+        private bool _ignoreSensitive;
+        public bool ShouldSerializeRelativeVirtualAddress() { return !_ignoreSensitive; }
```
Comment (nattress) [inline src/tools/r2rdump/R2RSection.cs]: Nit: You can write single-expression methods and properties as
`public bool ShouldSerializeRelativeVirtualAddress() => !_ignoreSensitive;`

**Comment by nattress** [comment]
We should be able to implement skipping properties of the serialized types conditionally without spreading the ignoreSensitive flag throughout the object model.  Also, the XML serializer uses a magic contract where the name of the serialized member can be preceded with "ShouldSerialize" that is not intuitive to readers. I had to go look it up because I thought none of these methods were getting called at first. We should be able to keep this skipping logic all contained within XmlDumper if we override the XmlIgnore attribute on fields we want to skip over. See https://docs.microsoft.com/en-us/dotnet/api/system.xml.serialization.xmlattributes.xmlignore?view=netframework-4.7.2 for an example. We would end up with a list of types and the fields we want to skip in one place during XmlSerializer construction. The `ignoreSensitive` flag would be passed directly from main to the XmlDumper.

File: src/tools/r2rdump/XmlDumper.cs [hunk 2]
```diff
@@ -25,6 +28,23 @@ public XmlDumper(R2RReader r2r, TextWriter writer, bool raw, bool header, bool d
             _unwind = unwind;
             _gc = gc;
             _sectionContents = sectionContents;
+
+            _ignoredProperties = new XmlAttributeOverrides();
```
Comment (nattress) [inline src/tools/r2rdump/XmlDumper.cs]: Currently this will always ignore those properties. Did you mean
`attrs.XmlIgnore = ignoreSensitive;` ?

Comment (acmyu) [reply src/tools/r2rdump/XmlDumper.cs]: oops, I'll fix that

**Comment by acmyu** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test please

**Comment by acmyu** [comment]
@dotnet-bot test OSX10.12 x64 Checked CoreFX Tests please

**Comment by acmyu** [comment]
@dotnet-bot test OSX10.12 x64 Checked CoreFX Tests please

**Comment by acmyu** [comment]
@dotnet-bot test OSX10.12 x64 Checked CoreFX Tests please

**Comment by acmyu** [comment]
@dotnet-bot test OSX10.12 x64 Checked CoreFX Tests please

**Comment by acmyu** [comment]
@dotnet-bot test Ubuntu x64 Checked CoreFX Tests please

**Pull request merged**